test/integration/src/test/java/com/liferay/faces/portal/test/integration/demo/PrimeFacesUsersPortletTester.java

```diff
@@ -48,506 +52,621 @@
-		browserStateAsserter.assertElementDisplayed(fileUploadChooserXpath + "/..");
+			// 2. Click the hidden *Reset Users* button to reset the state of the user data in preparation
+			// for testing.
+			WebElement hiddenResetUsersButton = browserDriver.findElementByXpath(
+					"//button[contains(@id,':hiddenResetUsersButton')]");
+			browserDriver.executeScriptInCurrentWindow("arguments[0].click();", hiddenResetUsersButton);
 
-		// Prior to attempting to upload an image, note the height of the photo that is displayed.
-		String portraitXpath = "//div/img[contains(@id,':portrait')]";
-		WebElement imgWebElement = browserDriver.findElementByXpath(portraitXpath);
-		Dimension imageSizeBefore = imgWebElement.getSize();
+			if (!screenNameCells.isEmpty()) {
+				browserDriver.waitFor(ExpectedConditions.stalenessOf(screenNameCells.get(0)));
+			}
 
-		// Click on the "Choose" button and select the file named "liferay-jsf-jersey.png" for upload.
-		WebElement fileUploadChooser = browserDriver.findElementByXpath(fileUploadChooserXpath);
+			browserDriver.waitForElementDisplayed(SCREEN_NAME_CELL_XPATH);
 
-		// TECHNICAL NOTE:
-		// Set PrimeFaces p:fileUpload transform style to "none" since it causes the element to not be displayed
-		// according to Selenium (although the element is visible to users).
-		browserDriver.executeScriptInCurrentWindow("arguments[0].style.transform = 'none';", fileUploadChooser);
-		fileUploadChooser.sendKeys(TestUtil.JAVA_IO_TMPDIR + "liferay-jsf-jersey.png");
-		browserDriver.waitFor(ExpectedConditions.stalenessOf(imgWebElement));
-		browserStateAsserter.assertElementDisplayed(portraitXpath);
-		imgWebElement = browserDriver.findElementByXpath(portraitXpath);
+			// 3. Click the *Preferences* link to navigate to edit mode.
+			String preferencesLinkXpath = "//a[contains(@id,':preferences')]";
+			browserDriver.clickElement(preferencesLinkXpath);
+
+			// 4. Enter "5" for the number of rows-per-page so that the pagination can be better tested with a small
+			// data set.
+			String rowsPerPageInputXpath = "//input[contains(@id,':rowsPerPage')]";
+			browserDriver.waitForElementEnabled(rowsPerPageInputXpath);
+			browserDriver.clearElement(rowsPerPageInputXpath);
+			browserDriver.sendKeysToElement(rowsPerPageInputXpath, "5");
 
-		// Verify that the height of the displayed photo is different than the height noted earlier.
-		Dimension imageSizeAfter = imgWebElement.getSize();
-		Assert.assertNotEquals(imageSizeBefore.getHeight(), imageSizeAfter.getHeight());
+			// 5. Click the *Submit* button to save the rows-per-page preference.
+			browserDriver.clickElement("//input[contains(@id,':submit')]");
+			browserDriver.waitForElementDisplayed(SCREEN_NAME_CELL_XPATH);
+			browserDriver.setWaitTimeOut(TestUtil.getBrowserDriverWaitTimeOut());
+			isStateReset = true;
+		}
+	}
+
+	private boolean areImageSizesEqual(Dimension expectedDimensions, Dimension actualDimensions) {
```

Comment:
@stiemannkj1 Technically there is no need to compare the width since it will always be "100" because of `style="width:100px;"` hard coded in users.xhtml. That's why we only compared height previously.